staging/src/k8s.io/client-go/util/csaupgrade/upgrade.go

```diff
@@ -88,16 +148,16 @@ func UpgradeManagedFields(
 		if !managerExists {
 			// There are no CSA managers that need to be converted. Nothing to do
 			// Return early
-			return nil
+			return nil, nil
```

Comment:
so if there's nothing to do, you return nil (this function is supposed to return the new `ManagedFieldsEntry`) as if the managed fields needed to be dropped? And in `UpgradeManagedFieldsPatch` you set that directly into the patch, so would you sometime create a patch that deletes managedFields?

Reply:
I appreciate you writing this eagle-eyed comment in the form of a question xD. I'll have to add a new test case and handle this case correctly in `UpgradeManagedFieldsPatch`

Reply:
fixed